From 83c25d0f3c5a65fca2c790ecb630c5f8c6b359e0 Mon Sep 17 00:00:00 2001
From: Chris Manton <cmanton@google.com>
Date: Mon, 7 Feb 2022 17:39:06 -0800
Subject: [PATCH 1/2] Security fix for potential OOB read in L2CAP

Bug: 212694559
Tag: #security
Test: gd/cert/run

Ignore-AOSP-First: Security fix
Change-Id: I6b5580a48295911f41e131e2328a3e284daeb68f
(cherry picked from commit 7f619cedc140b90faa290dbbc14d9c272fa4c6e6)
Merged-In:I6b5580a48295911f41e131e2328a3e284daeb68f
---
 stack/l2cap/l2c_ble.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/stack/l2cap/l2c_ble.cc b/stack/l2cap/l2c_ble.cc
index b826dc19c..16454a5b0 100644
--- a/stack/l2cap/l2c_ble.cc
+++ b/stack/l2cap/l2c_ble.cc
@@ -811,6 +811,11 @@ void l2cble_process_sig_cmd(tL2C_LCB* p_lcb, uint8_t* p, uint16_t pkt_len) {
 
     case L2CAP_CMD_CREDIT_BASED_RECONFIG_RES: {
       uint16_t result;
+      if (p + sizeof(uint16_t) > p_pkt_end) {
+        android_errorWriteLog(0x534e4554, "212694559");
+        LOG(ERROR) << "invalid read";
+        return;
+      }
       STREAM_TO_UINT16(result, p);
 
       L2CAP_TRACE_DEBUG(
-- 
2.33.1

